src/NetTopologySuite.IO.Esri/Extensions/ShpExtensions.cs

```diff
@@ -0,0 +1,120 @@
+﻿using NetTopologySuite.Geometries;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using NetTopologySuite.IO.Shapefile.Core;
+
+namespace NetTopologySuite.IO.Shapefile
+{
+    internal static class ShpExtensions
+    {
+        public static Point ToPoint(this ShpCoordinates shpPoint, bool hasZ, bool hasM)
+        {
+            if (hasZ && hasM)
+            {
+                return new Point(shpPoint.X, shpPoint.Y, shpPoint.Z) { M = shpPoint.M };
+            }
+            else if (hasZ)
+            {
+                return new Point(shpPoint.X, shpPoint.Y, shpPoint.Z);
+            }
+            else if (hasM)
+            {
+                return new Point(shpPoint.X, shpPoint.Y) { M = shpPoint.M };
+            }
+            else
+            {
+                return new Point(shpPoint.X, shpPoint.Y);
+            }
+        }
+
+
+        public static void SetCoordinates(this CoordinateSequence sequence, int index, ShpCoordinates coords, bool hasZ, bool hasM)
```

Comment:
This method needs a `PrecisionModel` parameter to make x- and y-coordinates precise
```C#
            sequence.SetX(index, pm.MakePrecise(coords.X));
            sequence.SetY(index, pm.MakePrecise(coords.Y));
```